benchmarks/src/main/java/org/elasticsearch/benchmark/xcontent/FilterContentBenchmark.java

```diff
@@ -0,0 +1,182 @@
+                sourceFile = "monitor_cluster_stats.json";
+                break;
+            case "index_stats":
+                sourceFile = "monitor_index_stats.json";
+                break;
+            case "node_stats":
+                sourceFile = "monitor_node_stats.json";
+                break;
+            default:
+                throw new IllegalArgumentException("Unknown type [" + type + "]");
+        }
+        source = readSource(sourceFile);
+
+        Map<String, Object> flattenMap = Maps.flatten(XContentHelper.convertToMap(source, true, XContentType.JSON).v2(), false, true);
+        Set<String> keys = flattenMap.keySet();
+        switch (fieldCount) {
+            case "10_field": {
+                AtomicInteger count = new AtomicInteger();
+                Set<String> filterKeys = keys.stream()
+                    .filter(key -> count.getAndIncrement() % 5 == 0)
+                    .limit(10)
+                    .collect(Collectors.toSet());
+                filters = filterKeys;
+            }
+                break;
+            case "half_field": {
+                AtomicInteger count = new AtomicInteger();
+                Set<String> halfKeys = keys.stream().filter(key -> count.getAndIncrement() % 2 == 0).collect(Collectors.toSet());
+                filters = halfKeys;
+            }
+                break;
+            case "all_field": {
+                Set<String> allKeys = new HashSet<>(keys);
+                allKeys.remove("cluster_uuid");
+                filters = allKeys;
+            }
+                break;
+            case "wildcard_field": {
+                Set<String> wildcardField = new HashSet<>(Arrays.asList("*stats"));
+                filters = wildcardField;
+            }
+                break;
+            case "10_wildcard_field": {
+                filters = new HashSet<>(
+                    Arrays.asList(
+                        "*stats.nodes*",
+                        "*stats.ind*",
+                        "*sta*.shards",
+                        "*stats*.xpack",
+                        "*stats.*.segments",
+                        "*stat*.*.data*",
+                        inclusive ? "*stats.**.request_cache" : "*stats.*.request_cache",
+                        inclusive ? "*stats.**.stat" : "*stats.*.stat",
+                        inclusive ? "*stats.**.threads" : "*stats.*.threads",
+                        "*source_node.t*"
+                    )
+                );
+            }
+                break;
+            default:
+                throw new IllegalArgumentException("Unknown type [" + type + "]");
+        }
+
+        Set<String> includes;
+        Set<String> excludes;
+        if (inclusive) {
+            includes = filters;
+            excludes = null;
+        } else {
+            includes = null;
+            excludes = filters;
+        }
+        parserConfig = XContentParserConfiguration.EMPTY.withFiltering(includes, excludes);
```

Comment:
Could you pull this into a method so it's super obvious that it's the same when you build it up front vs on the fly? It just helps with my copy-and-paste blindness.

Reply:
done